src/main/java/dbdr/security/service/BaseAuthenticationProvider.java

```diff
@@ -0,0 +1,42 @@
+package dbdr.security.service;
+
+import dbdr.global.exception.ApplicationError;
+import dbdr.global.exception.ApplicationException;
+import dbdr.security.dto.BaseUserDetails;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@RequiredArgsConstructor
+@Slf4j
+public class BaseAuthenticationProvider implements AuthenticationProvider {
+
+    private final BaseUserDetailsService baseUserDetailsService;
+
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+
+        BaseUserDetails unAuthUser = (BaseUserDetails) authentication.getPrincipal();
+        BaseUserDetails authUser = baseUserDetailsService.loadUserByUsernameAndRole(unAuthUser.getUserLoginId(), unAuthUser.getRole());
+
+        log.debug("unAuthUser : 검사시작 {}", unAuthUser.getUserLoginId());
+        //비밀번호 일치 확인
+        if (!passwordEncoder.matches(unAuthUser.getPassword(), authUser.getPassword())) {
+            throw new ApplicationException(ApplicationError.PASSWORD_NOT_MATCH);
+        }
```

Comment:
별도의 메서드로 분리하면 한눈에 로직을 이해할 수 있을 것 같습니다.
```suggestion
        checkPassword(unAuthUser.getPassword(), authUser.getPassword())
```

Reply:
반영하겠습니다!